distsql/request_builder.go

```diff
@@ -235,6 +236,15 @@ func (builder *RequestBuilder) SetFromSessionVars(sv *variable.SessionVars) *Req
 		builder.Request.SchemaVar = sv.TxnCtx.SchemaVersion
 	}
 	builder.txnScope = sv.TxnCtx.TxnScope
+	builder.IsStaleness = sv.TxnCtx.IsStaleness
+	if builder.IsStaleness && builder.txnScope != oracle.GlobalTxnScope {
```

Comment:
Why doesn't global scope support stale read?

Reply:
`global scope` also support stale read. If the txnScope is `local` here, we should pick `store` which match the label.

Reply:
So is it possible to also replace `build.txnScope` with `builder.MatchStoreLabels`?

Reply:
I think it is unnecessary, `build.txnScope` is also responsible for `(builder *RequestBuilder) verifyTxnScope()`.